src/client/app/components/visual-group/CreateVisualGroupComponent.tsx

```diff
@@ -0,0 +1,851 @@
+import { selectAllGroups, groupsApi } from '../../redux/api/groupsApi';
+
+/**
+ *   Visual graph component that shows the relationship between all groups and meters
+ * entered by an admin
+ * @returns D3 force graph visual
+ */
+interface CreateVisualGroupProps {
+    groups: GroupData[];
+    meters: MeterData[];
+}
+
+//Declare node types for groups and meters, will determine every node's design schema on the graphic
+type GroupNodeType = 'unselectedGroup' | 'selectedGroup' | 'childGroup' | 'deepGroup';
+type MeterNodeType = 'meter' | 'childMeter' | 'deepMeter';
+type AllNodeType = GroupNodeType | MeterNodeType;
+
+export const CreateVisualGroupComponent: React.FC<CreateVisualGroupProps> = ({
+    groups,
+    meters
+}) => {
+
+    const [selectedGroupId, setSelectedGroupId] = useState<number | null>(null);
+
+    const intl = useIntl();
+
+    /*Get Meter data from redux*/
+    const allMeters: MeterData[] = useAppSelector(selectAllMeters);
+    
+    /* Fetch deep groups data from API */
+    const { data: deepGroupsData = [], isLoading: deepGroupsLoading } = groupsApi.useGetAllGroupsDeepGroupsQuery();
+
+    /* Get all Group data from Redux */
+    const allGroups: GroupData[] = useAppSelector(selectAllGroups);
+    
+    /* Merge deepGroups data with allGroups */
+    const mergedGroups: GroupData[] = allGroups.map(group => {
+        const deepGroupData = deepGroupsData.find(dg => dg.id === group.id);
+        return {
+            ...group,
+            deepGroups: deepGroupData?.deepGroups || []
+        };
+    });
+
+
+    const selectedGroup: GroupData | undefined = mergedGroups.find(group => group.id === selectedGroupId);
+
+    /*Get all meters who are in any way a children of a group */
+    const groupedMeterIds: Set<number> = new Set(
+        mergedGroups.flatMap(group => group.deepMeters)
+    );
+
+    /*Only keep meters who have a relationship with a group */
+    const groupedMeters: MeterData[] = allMeters.filter(meterData => groupedMeterIds.has(meterData.id));
+
+    // Sort meters to minimize distance between meters that share groups
+    const sortMetersByGroupRelationships = (meters: MeterData[]): MeterData[] => {
+        if (meters.length <= 1) return meters;
+
+
+        const sortedMeters: MeterData[] = [];
+        const usedMeterIds = new Set<number>();
+
+
+        // Start with the first meter
+        sortedMeters.push(meters[0]);
+        usedMeterIds.add(meters[0].id);
+
+
+
+        // Greedy algorithm: always pick the meter that shares the most groups with the last placed meter
+        while (sortedMeters.length < meters.length) {
+            const lastMeter = sortedMeters[sortedMeters.length - 1];
+            let bestNextMeter: MeterData | null = null;
+            let maxSharedGroups = -1;
+
+
+            // Find the meter that shares the most groups with the last placed meter
+            for (const meter of meters) {
+                if (usedMeterIds.has(meter.id)) continue;
```

Comment:
As in another comment, I think avoiding jumps is better (I can explain why if you want). Inverting the if for use around the rest of the for code would do that.